classes/webservice/WebserviceRequest.php

```diff
@@ -1622,6 +1622,11 @@ protected function saveEntityFromXml($successReturnCode)
                     }
                     $result = $object->{$objectMethod}();
                     if ($result) {
+                        $objectMethod = ($this->method == 'POST' ? 'add' : 'update');
```

Comment:
The objectMethod variable is already defined above :)

Reply:
Yes, but it changes depending on webserviceParameters, which varies by object, I wrote it to you ...

Reply:
I know, but you repeat the same condition. You maybe need to change the original variable name or reconsider to use the new `$objectMethod` variable 🤔 
Because if the `$objectMethod` is renamed and is now `updateTest` you're not able to properly hook it.
I'm more in favor of using the `$objectMethod` as it is

Reply:
@PierreRambaud  Let me summarize the idea.
we create a hook for object monitoring, which was created through webservice independent of internal processing:

1. general name 
result: one hook for all, in function parameters it is possible to find out if hook is called during add or update

2. specific name, we add only 'add' or 'update' to the object name.
result: two hooks, one for 'add' and one for 'update'

3. name completely generated by the object variable,.
result: there will be several hooks by which the names can change depending on the existence of a special function of the webservice object
example: 
the 'customer' object does not currently have any special function => hook 'add' and 'update'
the next time you add a special function 'addWs' and 'updateWs' or others 'xxxWs' => hook 'addWs' and 'updateWs' or 'xxxWs'
and the original hooks call code will not work, the name of hook will change

Which solution do you suggest?
According to my priorities the solution is 2, 1, 3

Reply:
Or your solution is 4 ?

Reply:
You should do something like this:
```suggestion
                    // Call alternative method for add/update
                    $event = $objectMethod = ($this->method == 'POST' ? 'add' : 'update');
                    if (!empty($this->resourceConfiguration['objectMethods'][$objectMethod])) {
                        $objectMethod = $this->resourceConfiguration['objectMethods'][$objectMethod];
                    }

                    $result = $object->{$objectMethod}();
                    Hook::exec(
                        'actionWebserviceObject' . get_class($object) . ucfirst($objectMethod),
                        [
                            'object' => $object,
                            'xml' => $this->_inputXml,
                            'result' => $result,
                        ]
                    );
```
And the Hook needs to be outside  the `$result`

Reply:
I don't understand why you added **$event** = $object Method . If you don't use **$event**. I assume the resulting hook should be. Hook :: exec (
                         'actionWebserviceObject'. get_class ($object). ucfirst (**$event**),
                         [
                             'object' => $ object,
                             'xml' => $ this -> _ inputXml,
                             'result' => $ result,
                         ]
                     );

I agree with this proposal.

Reply:
I update my comment, the $event needs to be on the Hook::exec . Sorry for the delay

Reply:
@PierreRambaud in the hook, you deleted function ucfirst ($event), the result is not camelCase